flag.go

```diff
@@ -406,18 +406,29 @@ func (f *FlagSet) Set(name, value string) error {
 	if !ok {
 		return fmt.Errorf("no such flag -%v", name)
 	}
+
 	err := flag.Value.Set(value)
 	if err != nil {
-		return err
+		var flagName string
+		if len(flag.Shorthand) > 0 && len(flag.ShorthandDeprecated) == 0 {
+			flagName = fmt.Sprintf("-%s, --%s", flag.Shorthand, flag.Name)
+		} else {
+			flagName = fmt.Sprintf("--%s", flag.Name)
+		}
+		return fmt.Errorf("invalid argument %q for %q flag: %v", value, flagName, err)
```

Comment:
I'm trying to decide if `invalid argument` is always correct.  Maybe just `unable to set argument`